packages/react-reconciler/src/ReactFiberUnwindWork.js

```diff
@@ -160,113 +147,83 @@ function throwException(
     // This is a thenable.
     const thenable: Thenable = (value: any);
 
-    // TODO: Should use the earliest known expiration time
-    const currentTime = requestCurrentTime();
-    const expirationTimeMs = expirationTimeToMs(renderExpirationTime);
-    const currentTimeMs = expirationTimeToMs(currentTime);
-    const startTimeMs = expirationTimeMs - 5000;
-    let elapsedMs = currentTimeMs - startTimeMs;
-    if (elapsedMs < 0) {
-      elapsedMs = 0;
-    }
-    const remainingTimeMs = expirationTimeMs - currentTimeMs;
-
     // Find the earliest timeout of all the timeouts in the ancestor path.
-    // TODO: Alternatively, we could store the earliest timeout on the context
-    // stack, rather than searching on every suspend.
+    // TODO: We should store the earliest timeout on the context stack, rather
+    // than traversing the return path on every suspend.
```

Comment:
TODO: Don't need to store this on the stack. We only hit this path if we're IO-bound. Using the stack would slow down even the non-IO-bound case.